zap/src/main/java/org/parosproxy/paros/extension/option/OptionsViewPanel.java

```diff
@@ -75,15 +75,14 @@
 import org.parosproxy.paros.model.OptionsParam;
 import org.parosproxy.paros.view.AbstractParamPanel;
 import org.parosproxy.paros.view.WorkbenchPanel;
-import org.zaproxy.zap.extension.brk.BreakpointsParam;
 import org.zaproxy.zap.utils.FontUtils;
 import org.zaproxy.zap.utils.TimeStampUtils;
 import org.zaproxy.zap.utils.ZapNumberSpinner;
 import org.zaproxy.zap.view.LayoutHelper;
 
 // ZAP: 2011: added more configuration options
 
-@SuppressWarnings("serial")
+@SuppressWarnings({"serial", "deprecation"})
```

Comment:
The code should be changed to allow the new extension to affect the behaviour.

Reply:
how can I do that? using reflection? Have we done this anywhere else which I can refer to ?

Reply:
Add an interface that allows to set a boolean and means to (un)register it and change this code to use it.

Not exactly this case but very similar: https://github.com/zaproxy/zaproxy/pull/7080/files#diff-bf38de7e5173b1d2d2dee70bd4d43ced6eb5c9391b51ad96e3265aacc137480e

Reply:
okay got it